include/aws/common/array_list.inl

```diff
@@ -62,14 +62,15 @@ void aws_array_list_init_static(
     void *raw_array,
     size_t item_count,
     size_t item_size) {
-    AWS_FATAL_ASSERT(raw_array);
-    AWS_FATAL_ASSERT(item_count);
-    AWS_FATAL_ASSERT(item_size);
+    AWS_FATAL_PRECONDITION(list);
+    AWS_FATAL_PRECONDITION(raw_array);
+    AWS_FATAL_PRECONDITION(item_count);
+    AWS_FATAL_PRECONDITION(item_size);
```

Comment:
Should we raise an `AWS_ERROR_INVALID_ARGUMENT` error or keep the `AWS_FATAL_PRECONDITION`? I'd suggest to pick one and use for both `init_static` and `init_dynamic`

Reply:
@ColdenCullen

Reply:
The static init functions are void so I could not return the AWS_ERROR_INVALID_ARGUMENT error as in the dynamic versions. However, I could raise a AWS_FATAL_PRECONDITION in the dynamic functions.